src/GraphQL.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Gallerya\GraphQL.
+ */
+
+namespace Netzstrategen\Gallerya;
+
+/**
+ * GraphQL integration.
+ */
+class GraphQL {
+
+  /**
+   * @implements graphql_register_types
+   */
+  public static function graphql_register_types() {
+    register_graphql_connection(
+      [
+        'fromType' => 'ProductVariation',
+        'toType' => 'MediaItem',
+        'fromFieldName' => 'galleryImages',
+        'resolve' => function ($source, array $args, $context, $info) {
+          $variation_gallery_image_ids = Woocommerce::getVariationGalleryImages($source->databaseId);
+          if (empty($variation_gallery_image_ids)) {
+            return ['nodes' => []];
+          }
+          $resolver = new \WPGraphQL\Data\Connection\PostObjectConnectionResolver($source, $args, $context, $info, 'attachment');
+          $resolver->set_query_arg('post_type', 'attachment');
+          $resolver->set_query_arg('post__in', $variation_gallery_image_ids);
```

Comment:
Do we need to care for the publishing status of the attachments?

Reply:
No, not in this context.